network/consensus/gcpv2/core/round_controller.go

```diff
@@ -314,10 +313,12 @@ func (r *PhasedRoundController) _startFullRealm(prepWasSuccessful bool) {
 	r.roundWorker.SetTimeout(endOf)
 
 	inslogger.FromContext(r.realm.roundContext).Warnf(
-		"Starting consensus full realm: self={%v}, ephemeral=%v, unsafe=%v, startedAt=%v, endOf=%v, census=%+v", r.realm.GetLocalProfile(),
+		"Starting consensus full realm: self={%v}, ephemeral=%v, unsafe=%v, startedAt=%v, endOf=%v, census=%+v, timings=%s", r.realm.GetLocalProfile(),
 		r.realm.ephemeralFeeder != nil, r.realm.unsafeRound,
 		args.LazyTimeFmt("15:04:05.000000", r.realm.GetStartedAt()),
-		args.LazyTimeFmt("15:04:05.000000", endOf), active)
+		args.LazyTimeFmt("15:04:05.000000", endOf), active,
+		r.realm.timings.String(),
```

Comment:
Don't call `.String()` directly. It will be called lazily when message is building, after level checks.

Reply:
then it shows `timings={100ms 0s 0s 400ms 0s 1.2s 1.6s 1.8s}` instead `timings=EndOfConsensus: 1.8s, EndOfPhase1: 400ms, EndOfPhase2: 1.2s, EndOfPhase3: 1.6s"}`

Reply:
Yeah, it's because of pointer receiver